examples/webgpu_camera_logarithmicdepthbuffer.html

```diff
@@ -0,0 +1,353 @@
+
+			// Generate a number of text labels, from 1µm in size up to 100,000,000 light years
+			// Try to use some descriptive real-world examples of objects at each scale
+
+			const labeldata = [
+				{ size: .01, scale: 0.0001, label: 'microscopic (1µm)' }, // FIXME - triangulating text fails at this size, so we scale instead
+				{ size: .01, scale: 0.1, label: 'minuscule (1mm)' },
+				{ size: .01, scale: 1.0, label: 'tiny (1cm)' },
+				{ size: 1, scale: 1.0, label: 'child-sized (1m)' },
+				{ size: 10, scale: 1.0, label: 'tree-sized (10m)' },
+				{ size: 100, scale: 1.0, label: 'building-sized (100m)' },
+				{ size: 1000, scale: 1.0, label: 'medium (1km)' },
+				{ size: 10000, scale: 1.0, label: 'city-sized (10km)' },
+				{ size: 3400000, scale: 1.0, label: 'moon-sized (3,400 Km)' },
+				{ size: 12000000, scale: 1.0, label: 'planet-sized (12,000 km)' },
+				{ size: 1400000000, scale: 1.0, label: 'sun-sized (1,400,000 km)' },
+				{ size: 7.47e12, scale: 1.0, label: 'solar system-sized (50Au)' },
+				{ size: 9.4605284e15, scale: 1.0, label: 'gargantuan (1 light year)' },
+				{ size: 3.08567758e16, scale: 1.0, label: 'ludicrous (1 parsec)' },
+				{ size: 1e19, scale: 1.0, label: 'mind boggling (1000 light years)' }
+			];
+
+			init();
+
+			function init() {
+
+				if ( WebGPU.isAvailable() === false && WebGL.isWebGL2Available() === false ) {
+
+					document.body.appendChild( WebGPU.getErrorMessage() );
+
+					throw new Error( 'No WebGPU or WebGL2 support' );
+
+				}
+
+				container = document.getElementById( 'container' );
+
+				const loader = new FontLoader();
+				loader.load( 'fonts/helvetiker_regular.typeface.json', function ( font ) {
+
+					const scene = initScene( font );
+
+					// Initialize two copies of the same scene, one with normal z-buffer and one with logarithmic z-buffer
+					objects.normal = initView( scene, 'normal', false );
+					objects.logzbuf = initView( scene, 'logzbuf', true );
+
+				} );
+
+				stats = new Stats();
+				container.appendChild( stats.dom );
+
+				// Resize border allows the user to easily compare effects of logarithmic depth buffer over the whole scene
+				border = document.getElementById( 'renderer_border' );
+				border.addEventListener( 'pointerdown', onBorderPointerDown );
+
+				window.addEventListener( 'mousemove', onMouseMove );
+				window.addEventListener( 'resize', onWindowResize );
+				window.addEventListener( 'wheel', onMouseWheel );
+
+			}
+
+			function initView( scene, name, logDepthBuf ) {
+
+				const framecontainer = document.getElementById( 'container_' + name );
+
+				const camera = new THREE.PerspectiveCamera( 50, screensplit * SCREEN_WIDTH / SCREEN_HEIGHT, NEAR, FAR );
+				scene.add( camera );
+
+				const renderer = new WebGPURenderer( { antialias: true, logarithmicDepthBuffer: logDepthBuf } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( SCREEN_WIDTH / 2, SCREEN_HEIGHT );
+				renderer.setAnimationLoop( render );
```

Comment:
While migrating the examples towards `setAnimationLoop()` I have noticed an issue in this example. The  `render()` function loop _both_ renderers but since `setAnimationLoop()` is called twice, the frame rate is double as high as expected.

How are apps supposed to use more than one `WebGPURenderer` with a single global animation loop? 

AFAIK, if the example calls `setAnimationLoop()` only for a one renderer, the internal state of the second one isn't correct (since `NodeFrame.update()` isn't called and `renderer.info` isn't updated).

If the example uses an animation loop per renderer, which animation loop does it use to update its "global state"? No matter which one you use, the second animation loop might use frame-late data.

Maybe we need a way to trigger the update logic in the `Animation` class from outside somehow...

Reply:
I'm also battling a bit with this problem as I was planning to introduce `renderer.xr.setAnimationLoop()` (Quest Browser can have immersive + non immersive at the same time).

Where do the physics run? 🤔

We'll probably have to create an issue for this.

Reply:
> ..If the example uses an animation loop per renderer, which animation loop does it use to update its "global state"? No matter which one you use, the second animation loop might use frame-late data...

This should not be the expected result, `Animation` has a frame counter per Renderer, regardless of the user using `setAnimationLoop()` as would happen in `WebGLRenderer`, this ensures that the user can still use `requestAnimationFrame` without harming the events per frame in the `Nodes`.

> While migrating the examples towards setAnimationLoop() I have noticed an issue in this example. The render() function updates both renderers but since setAnimationLoop() is called twice, the frame rate is double as high as expected.

I think this is behavior is correct if two renderers depend on using `setAnimationLoop()`, both must be isolated, and this conflict of Node updates should not exist as we have an `Animation` and `NodeFrame` class per Renderer in `WebGPURenderer`. It could just use one `setAnimationLoop` or one `requestAnimationFrame` for multiples Renderers, if this doesn't work it could be a bug.

Although it is an unusual approach to use more than one device, practically no engine duplicates devices, perhaps the user could solve this with `RenderTargets`, scissor, etc. if the renderer is working and declared in the same thread, in case of different threads having independent `setAnimationLoop` should be more useful.

> I'm also battling a bit with this problem as I was planning to introduce renderer.xr.setAnimationLoop() (Quest Browser can have immersive + non immersive at the same time).

Shouldn't `Renderer.setAnimationLoop()` manage these calls internally, and we only have one function for this event?

Reply:
> This should not be the expected result, Animation has a frame counter per Renderer, regardless of the user using setAnimationLoop() as would happen in WebGLRenderer, this ensures that the user can still use requestAnimationFrame without harming the events per frame in the Nodes.

Sorry for not being precise here: Consider two renderers share a single scene. If you want to update objects and then render the scene, you could do this before:

```js
function animate() {

    requestAnimationFrame( animate );

    object.position.x += 1;

    renderer1.render( scene, camera );
    renderer2.render( scene, camera );

}
```
AFAICS, the following code is invalid since `renderer2` does not use `setAnimationLoop()`.
```js
renderer1.setAnimationLoop( animate );

function animate() {

    object.position.x += 1;

    renderer1.render( scene, camera );
    renderer2.render( scene, camera );

}
```
So let's rewrite the code to:
```js
renderer1.setAnimationLoop( animate1 );
renderer2.setAnimationLoop( animate2 );

function animate1() {

    object.position.x += 1;

    renderer1.render( scene, camera );

}

function animate2() {

    // object.position.x += 1;

    renderer2.render( scene, camera );

}
```
In which one do you update the object? You never know when the animation is scheduled by the browser so `animate2()` might run before `object` has been updated (frame-late effect). 

Maybe we need something like this?
```js
function animate() {

    requestAnimationFrame( animate );

    renderer1.update();
    renderer2.update();

    object.position.x += 1;

    renderer1.render( scene, camera );
    renderer2.render( scene, camera );

}
```

Reply:
> Since Animation isn't used now, how is info.reset() called? Does the app needs to do this?

I was considreing the user customize the `Info` if needed, so he could create new classes for this https://github.com/mrdoob/three.js/pull/27889#issuecomment-1990164451

What do you think about share the info between renderers?

```js
const info  = new Info(); // it can be a customized Info class too

renderer1.info = info;
renderer2.info = info;
```

> Is the missing call in nodeFrame.update() in Animation an issue? Does the app has to call this method as well?

It continued to be updated, `Animation` class of  `WebGPURenderer` have a `requestAnimationFrame()` internal to sync the frame events, it's canceled if the user dispose the renderer.

https://github.com/mrdoob/three.js/blob/32d19ebeef7a5a3b7a86a4865df3f883c050e88f/examples/jsm/renderers/common/Animation.js#L17-L29

Reply:
Um, my feeling is the update loop in `Animation` should _only_ run if the user does use `setAnimationLoop()`.

If the user manages the animation loop on app level and not via the renderer, the user should tell the renderer when to update its internal frame counter, the timer and other frame related components. 

A separate `Info` object which is managed on app level and injected into the renderer (or multiple renderers) could be one solution.